gap/PolygonalComplexes/properties.gd

```diff
@@ -354,7 +355,106 @@ DeclareAttribute( "EdgeCounter", IsTwistedPolygonalComplex );
 #! @Returns a list of pairs
 #! @Arguments complex
 DeclareAttribute( "FaceCounter", IsTwistedPolygonalComplex );
+#! @EndGroup
+
+#! @BeginGroup ButterflyCounter
+#! @Description
+#! Return the <E>butterfly counter</E> of the given simplicial surface.
+#! The butterfly counter is a list of pairs
+#! <E>[[degList1,degList2],number]</E>, where <E>number</E> counts the
+#! number of butterflies whose vertex degrees match
+#! <E>[degList1,degList2]</E>, whereby <E>degList1</E> denotes the
+#! vertex degree of the vertices that are incident to the edge
+#! inducing the corresponding butterfly and <E>degList2</E> contains
+#! the vertex degrees of the two remaining vertices of the butterfly.
+#!
+### ! . Given an inner edge <E>e</E>,
+### ! the first list contains the vertex degrees of the vertices incident
+### ! to <E>e</E>. Since an inner edge gives rise to a butterfly the
+### ! second list contains the vertex degrees of the remaining two vertices
+### ! of the butterfly. The integer <E>number</E> counts the number of
+### ! butterflies whose vertex degrees match <E>list</E>.
+#!
+#! As an example, consider the double-5-gon:
+#! <Alt Only="TikZ">
+#!      \input{Image_Double5gon.tex}
+#! </Alt>
+#! @ExampleSession
+#! gap> s:=AllSimplicialSpheres(10)[1];
+#! simplicial surface (7 vertices, 15 edges, and 10 faces)
+#! gap> ButterflyCounter(s);
+#! [ [ [ [ 4, 4 ], [ 5, 5 ] ], 5 ], [ [ [ 4, 5 ], [ 4, 4 ] ], 10 ] ]
+#! @EndExampleSession
+#!
+#! @Returns a list of pairs
+#! @Arguments surface
+DeclareAttribute( "ButterflyCounter", IsSimplicialSurface);
+#! @EndGroup
+
+#! @BeginGroup UmbrellaCounter
+#! @Description
+#! Return the <E>umbrella counter</E> of the given simplicial surface.
+#! The umbrella counter is a list of pairs <E>[degreeLlist, number]</E>,
+#! where <E>number</E> counts the number of umbrellas whose vertex degrees
+#! of vertices contained at the boundary of the umbrella match
+#! <E>degreeList</E>, i.e. for every vertex contained in the boundary
+#! of an umbrella there is exactly one entry of <E>degreeList</E> such that
+#! the vertex is incident to this number of faces.
+
+#! The different lists in <E>degreeList</E> are always sorted but may
+#! contain duplicates.
+#!
+#! As an example, consider the double-6-gon:
+#! <Alt Only="TikZ">
+#!      \input{Image_Double6gon.tex}
+#! </Alt>
+#! @ExampleSession
+#! gap> s:=AllSimplicialSpheres(12)[1];
+#! simplicial surface (8 vertices, 18 edges, and 12 faces)
+#! gap> UmbrellaCounter(s);
+#! [ [ [ 4, 6, 4, 6 ], 6 ], [ [ 4, 4, 4, 4, 4, 4 ], 2 ] ]
+#! @EndExampleSession
+#!
+#! @Returns a list of pairs
+#! @Arguments surface
+DeclareAttribute( "UmbrellaCounter", IsSimplicialSurface);
+#! @EndGroup
 
+#! @BeginGroup ThreeFaceCounter
+#! @Description
+#! Return the <E>three-face counter</E> of the given simplicial surface.
+#! The three-face counter is a list of pairs <E>[degreeLlist, number]</E>,
```

Comment:
Here is a typo and better is maybe: "The three-face counter is a list of
pairs [degreeList, number], where degreeList is a triple..."